src/core/layertree/qgslayertreeutils.cpp

```diff
@@ -282,12 +281,30 @@ bool QgsLayerTreeUtils::layersModified( const QList<QgsLayerTreeLayer *> &layerN
   const auto constLayerNodes = layerNodes;
   for ( QgsLayerTreeLayer *layerNode : constLayerNodes )
   {
-    QgsVectorLayer *vl = qobject_cast<QgsVectorLayer *>( layerNode->layer() );
-    if ( !vl )
+    QgsMapLayer *layer = layerNode->layer();
+    if ( !layer )
       continue;
 
-    if ( vl->isEditable() && vl->isModified() )
-      return true;
+    switch ( layer->type() )
+    {
+      case QgsMapLayerType::VectorLayer:
+      {
+        QgsVectorLayer *vl = qobject_cast<QgsVectorLayer *>( layer );
+        if ( vl->isEditable() && vl->isModified() )
+          return true;
+      }
+      break;
+      case QgsMapLayerType::MeshLayer:
+      {
+        QgsMeshLayer *ml = qobject_cast<QgsMeshLayer *>( layer );
+        if ( ml->isEditable()  && ml->isFrameModified() )
+          return true;
```

Comment:
Again, ideally this special handling would be avoidable if some methods are promoted to QgsMapLayer class instead